lib/hamt.ml

```diff
@@ -513,6 +524,18 @@ module Make (Config : CONFIG) (Key : Hashtbl.HashedType) :
 
   let bindings hamt = fold (fun k v acc -> (k, v) :: acc) hamt []
 
+  let to_seq t = fold (fun k v acc () -> Seq.Cons ((k, v), acc)) t Seq.empty
+
+  let of_seq =
+    let rec loop seq acc =
+      match seq () with
+      | Seq.Nil -> acc
+      | Cons ((k, v), seq) ->
+          let acc = add k v acc in
+          loop seq acc
+    in
+    fun seq -> loop seq empty
+
```

Comment:
This looks correct and I'm fine with that implementation. However, you could also consider, drawing inspiration from the Import submodule, an expected faster one that uses mutation under the hood. The idea is to start from an empty hamt, then for each added element, instead of copying it's ancestor hierarchy, only mutate its parent to add the element.

The point is that losing persistence of the temporary constructions is less of an issue if we're only interested in the final result (which is the expectation of a function named `of_seq`). However it plays less nicely with concurrency, and probably some other low-level nasty tricks.

I believe I had a benchmark for this exact comparison, with lists instead of sequences. 

http://gallium.inria.fr/blog/implementing-hamt-for-ocaml/ gives (a very few) more details.

Reply:
Using add_mut now.